tests/OSPSuite.Core.Tests/Domain/ObjectPathSpecs.cs

```diff
@@ -116,6 +117,39 @@ public void should_return_the_desired_entity()
       }
    }
 
+   public class When_resolving_an_object_from_an_entity_with_parent_path : When_resolving_an_object_from_an_absolute_object_path_regarding_the_ref_entity
+   {
+      protected override void Context()
+      {
+         base.Context();
+         Random random = new Random();
+         int n = random.Next(1, 11);
+         var lstPaths = Enumerable.Range(0, n)
+            .Select(_ => GenerateRandomString(random, 8))
+            .ToList();
+
+         _organ.RootContainer.ParentPath = new ObjectPath(lstPaths);
+         lstPaths.Reverse();
+         foreach (var path in lstPaths)
+         {
+            sut.AddAtFront(path);
+         }
+      }
+
+      [Observation]
+      public void should_return_the_desired_entity()
+      {
+         sut.Resolve<IParameter>(_organ).ShouldBeEqualTo(_parameter);
+      }
+
+      private static string GenerateRandomString(Random random, int length)
```

Comment:
I have no idea why we need to generate random strings suddenly in a text. I have no idea what's going on.
Please create a path that we can analyze visually and not in a weird loop.